internal/lsp/server.go

```diff
@@ -608,6 +611,46 @@ func (l *LanguageServer) handleTextDocumentInlayHint(
 	return inlayHints, nil
 }
 
+func (l *LanguageServer) handleTextDocumentDefinition(
+	_ context.Context,
+	_ *jsonrpc2.Conn,
+	req *jsonrpc2.Request,
+) (result any, err error) {
+	var params types.DefinitionParams
+	if err := json.Unmarshal(*req.Params, &params); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal params: %w", err)
+	}
+
+	contents, ok := l.cache.GetFileContents(params.TextDocument.URI)
+	if !ok {
+		return nil, fmt.Errorf("failed to get file contents for uri %q", params.TextDocument.URI)
+	}
+
+	orc := oracle.New()
+	query := oracle.DefinitionQuery{
+		Filename: uri.ToPath(l.clientIdentifier, params.TextDocument.URI),
+		Pos:      positionToOffset(contents, params.Position),
+		Modules:  l.cache.GetAllModules(),
+		Buffer:   []byte(contents),
+	}
+
+	definition, err := orc.FindDefinition(query)
+	if err != nil {
+		// fail silently — the user could have clicked anywhere
+		return nil, nil //nolint:nilerr,nilnil
```

Comment:
I thought we could return `struct{}{}` here in such cases?

Reply:
I had some problems before returning an empty array, but yeah that seems to work. I'll update.